### PR TITLE
Feature: Resources Page

### LINKED ISSUE #196: Write GROQ query to pull resources and organize them into their categories
The content in Sanity is organized into types one can create called schemas. (I suppose they can be thought of as tables in a database, although they're not exactly the same of course.) To query the content stored in a schema, Sanity has its own query language called [GROQ](https://www.sanity.io/docs/groq-reference), which one can experiment with locally on `http://localhost:3333/vision`.

For our resources page, there are two schemas involved: one is [the resource schema](https://github.com/HackAtUCI/irvinehacks-site-2024/blob/main/apps/sanity/schemas/resource.ts), and the other is [the resource category schema](https://github.com/HackAtUCI/irvinehacks-site-2024/blob/main/apps/sanity/schemas/resourceCategory.ts). There is a field called "Resource Category" in the resource schema that references the category schema, which means when querying a resource, it will contain a "reference" to the specific category that was chosen, instead of the content of the category itself.

To query content, the GROQ query in [this file](https://github.com/HackAtUCI/irvinehacks-site-2024/blob/main/apps/site/src/app/(main)/resources/getResources.ts) should be modified. It uses pulls from the `resource` schema (ex. `*[_type == "resource"]`), but leaves the references as references, which leaves the category content inaccessible. I'll leave it to you to figure out how to get the resource category information from this query (see [GROQ joins](https://www.sanity.io/docs/groq-joins)).

Once the resources have been pulled along with information related to their categories, please organize them into their respective categories (perhaps in a hashmap where the key is the title of the category).

### PR DESCRIPTION
**Pull request by alexanderl19**
closes #196 #197

### CODE REVIEW

File: apps/site/src/app/(main)/resources/page.tsx [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+
+export const metadata: Metadata = {
+	title: "Resources | IrvineHacks 2024",
+};
+
+export default async function Schedule() {
+	if (process.env.MAINTENANCE_MODE_SCHEDULE) {
+		redirect("/");
+	}
+
```
Comment (samderanova) [inline apps/site/src/app/(main)/resources/page.tsx:16]: Issue: Could you rename the function to `Resources` and then change the variable name to `MAINTENANCE_MODE_RESOURCES`?

File: apps/site/src/app/(main)/resources/page.tsx [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+									<p>{description}</p>
+								</div>
+								<div className="flex flex-col gap-3 lg:gap-4">
+									{resources.map(({ _id, title, link }) => (
+										<Button
+											key={_id}
+											text={title}
+											href={link}
+											className="w-[100%!important]"
+										/>
```
Comment (samderanova) [inline apps/site/src/app/(main)/resources/page.tsx:49]: Issue: Could we center the button text and alternate button color backgrounds, as in the Figma?

**Review by samderanova — CHANGES_REQUESTED** [review]
Great start so far! A few things to note:

File: apps/site/src/app/(main)/resources/page.tsx (hunk 2 above)
Comment (alexanderl19) [reply apps/site/src/app/(main)/resources/page.tsx]: I just pushed a commented out prop into the button component that alternates the button color. I'll leave it up to you / other reviewers to decided whether to modify the button component in this PR or another. Feel free to push changes to my branch.

Comment (samderanova) [reply apps/site/src/app/(main)/resources/page.tsx]: We can refactor the Button component in another issue, as the Button component is getting a bit messy.

File: apps/site/src/app/(main)/resources/page.tsx [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+								className="max-w-5xl w-full mx-auto mb-12 bg-[var(--color-cream)] text-[#2F1C00] p-12 rounded-2xl lg:grid lg:gap-20 lg:grid-cols-2"
+							>
+								<div>
+									<div className="flex gap-2 lg:flex-col">
+										{/* eslint-disable-next-line @next/next/no-img-element */}
+										<img className="w-8 h-auto lg:w-16" src={iconUrl} alt="" />
+										<h2 className="text-2xl font-bold">{title}</h2>
+									</div>
+									<p>{description}</p>
+								</div>
```
Comment (samderanova) [inline apps/site/src/app/(main)/resources/page.tsx:41]: Sorry, one more thing. Could we make this `div` vertically centered?

**Review by samderanova — CHANGES_REQUESTED** [review]

**Review by samderanova — DISMISSED** [review]
Looks good for now. We can refactor the Button component in a separate issue. Thanks for the quick work on this!

**Comment by waalbert** [comment]
image

Resources is a bit off centered on my iPhone 13 mini Safari

**Review by samderanova — DISMISSED** [review]
Had to push one more small change to include the home and resources links in the navbar, but other than that, looks good!

**Review by waalbert — DISMISSED** [review]

**Review by samderanova — APPROVED** [review]

**Pull request merged**